### PR TITLE
add cross region vpc peering

### PR DESCRIPTION
**Pull request by mteiste**
fix issue #56

### CODE REVIEW

File: resource_vpc_peering_connection.go [hunk 1]
```diff
@@ -41,6 +41,18 @@ func resourceVPCPeeringConnection() *schema.Resource {
 				Required:    true,
 				Type:        schema.TypeString,
 			},
+			"peer_region": {
+				Description: "AWS region of the peered VPC (if not in the same region)",
```
Comment (rikonen) [inline resource_vpc_peering_connection.go]: If not in the same region as what.

File: resource_vpc_peering_connection.go [hunk 2]
```diff
@@ -41,6 +41,18 @@ func resourceVPCPeeringConnection() *schema.Resource {
...
+				ForceNew:    true,
+				Optional:    true,
+				Type:        schema.TypeString,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					if new != "" {
+						return false
+					}
+					return true
```
Comment (rikonen) [inline resource_vpc_peering_connection.go]: `return new == ""` would work as well I suppose

File: resource_vpc_peering_connection.go [hunk 3]
```diff
@@ -58,15 +70,34 @@ func resourceVPCPeeringConnection() *schema.Resource {
...
-			PeerVPC:          d.Get("peer_vpc").(string),
-		},
+	peerRegion := d.Get("peer_region").(string)
+
+	var (
+		pc  *aiven.VPCPeeringConnection
+		err error
 	)
 
+	if peerRegion != "" {
```
Comment (rikonen) [inline resource_vpc_peering_connection.go]: Only having single `Create` in the API client and something like

```
apiPeerRegion := nil
if peerRegion != "" {
    apiPeerRegion = &peerRegion
}
pc, err := client.VPCPeeringConnections.Create(...)
```

would seem more convenient.

File: resource_vpc_peering_connection.go (hunk 1 above)
Comment (mteiste) [reply resource_vpc_peering_connection.go]: fixed

File: resource_vpc_peering_connection.go (hunk 2 above)
Comment (mteiste) [reply resource_vpc_peering_connection.go]: fixed

**Comment by mteiste** [comment]
manually tested
- created vpc and one peering in the same region and one peering in an other region
- applied and verified from Aiven console that the vpc and peerings were created
- refreshed
- tested plan (to test that the same region peering will stay unchanged even though the refresh will fill the peer_region
- removed peerings from the configuration
- applied and verified from Aiven console that the peerings were deleted
- removed vpc from configuratoin
- applied and verified from Aiven console that the vpc status was changed to deleted

**Pull request merged**